File: timelineview/src/main/java/com/konatsup/timelineview/Track.java
package com.konatsup.timelineview;

import java.util.ArrayList;
import java.util.Map;

public class Track {
    private Map<Integer, Composition> compositionMap;

    /* constructor */
    public Track(Map<Integer, Composition> compositionMap){
        this.compositionMap = compositionMap;
    }

    /* getter and setter */
    public Map<Integer, Composition> getCompositionMap() {
        return compositionMap;
    }

    public void setCompositionMap(Map<Integer, Composition> compositionMap) {
        this.compositionMap = compositionMap;
    }

    /* functions */
    public ArrayList<Composition> getCompositionList() {
        return  (ArrayList<Composition>) this.compositionMap.values();
    }

    public void createComposition(int pos) {
        // TODO: select default composition
         Composition composition = new AudioComposition(0);
         this.compositionMap.put(pos, composition);
    }

    public void addComposition(int position, Composition composition) {
        this.compositionMap.put(position, composition);
    }

}

File: timelineview/src/main/java/com/konatsup/timelineview/Track.kt
package com.konatsup.timelineview

import java.util.ArrayList

class Track(var compositionMap: MutableMap<Int, Composition>) {

    /* functions */
    val compositionList: ArrayList<Composition>
        get() = this.compositionMap.values as ArrayList<Composition>

    fun createComposition(pos: Int) {
        // TODO: select default composition
        val composition = AudioComposition(0)
        this.compositionMap[pos] = composition
    }

    fun addComposition(position: Int, composition: Composition) {
        this.compositionMap[position] = composition
    }

}
